[18.0][MIG] product_route_profile: Migration to 18.0

StefanRijnhart opened the pull request:
Standard migration (applying Fields.Command where possible), but with an additional commit to improve backwards compatibility.

Kev-Roche committed:
[ADD] module: product_route_profile

Commit:
[UPD] Update product_route_profile.pot

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
product_route_profile 14.0.1.0.1

awisky committed:
[FIX] product_route_profile _inverse_route_ids store selected route_profile_id

OCA-git-bot committed:
product_route_profile 14.0.1.0.2

bosd committed:
Added translation using Weblate (Dutch)

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-14.0/stock-logistics-warehouse-14.0-product_route_profile
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-14-0/stock-logistics-warehouse-14-0-product_route_profile/

etobella committed:
*: Fix pylint

Commit:
[UPD] Update product_route_profile.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-14.0/stock-logistics-warehouse-14.0-product_route_profile
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-14-0/stock-logistics-warehouse-14-0-product_route_profile/

bosd committed:
[MIG] product_route_profile: Migration to 15.0

bosd committed:
[IMP] product_route_profile: add custom icon

generated with oca icon generator.

bguillot committed:
[FIX] product_route_profile: prevent triggering inverse method

Set route_ids as readonly to avoid triggering inverse method and changing
the route_profile_id when adding force_route_profile_id

StefanRijnhart committed:
[IMP] product_route_profile: pre-commit auto fixes

StefanRijnhart committed:
[MIG] product_route_profile: Migration to 17.0

Commit:
[UPD] Update product_route_profile.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-17.0/stock-logistics-warehouse-17.0-product_route_profile
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-17-0/stock-logistics-warehouse-17-0-product_route_profile/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (16 of 16 strings)

Translation: stock-logistics-warehouse-17.0/stock-logistics-warehouse-17.0-product_route_profile
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-17-0/stock-logistics-warehouse-17-0-product_route_profile/it/

StefanRijnhart committed:
[MIG] product_route_profile: Migration to 18.0

StefanRijnhart commented:
/ocabot migration product_route_profile

jbaudoux commented on product_route_profile/models/stock_route.py:
```diff
@@ -0,0 +1,54 @@
+        store=False,
+    )
+
+    def _compute_product_ids(self):
```
missing depends on route_profile_id.route_ids

StefanRijnhart replied:
Thanks! By adding a many2many inverse I could easily define the `api.depends`.

jbaudoux replied:
Great. You remove the know issue note then ?
:)

StefanRijnhart replied:
Done!

jbaudoux commented on stock_warehouse_resupply_route_push/tests/test_stock_warehouse_resupply_route_push.py:
```diff
@@ -257,6 +257,7 @@ def test_05_recreate_routes_assignation(self):
         wh_4.route_ids += r_wh_3_to_wh4
         # Change to push and recreate routes:
         wh_4.write({"resupply_wh_push": True})
+        self.env.invalidate_all()  # product_route_profile compatibility
```
This should not occur here. There is a missing depend for cache invalidation or the invalidation should occur in the module that requires it, but not here.

StefanRijnhart replied:
Thanks, also solved by adding the compute method depends.

StefanRijnhart committed:
[IMP] product_route_profile: improve backwards compatibility

Fixes
```
  File "/__w/stock-logistics-warehouse/stock-logistics-warehouse/stock_warehouse_resupply_route_push/tests/test_stock_warehouse_resupply_route_push.py", line 271, in test_05_recreate_routes_assignation
    self.assertIn(new_wh_1_to_wh4, self.product.route_ids)
AssertionError: stock.route(69,) not found in stock.route(54,)
```

jbaudoux commented:
/ocabot merge nobump
